#VRML_SIM R2020a utf8
WorldInfo {
  basicTimeStep 16
}
Viewpoint {
  orientation 0.22176285237218632 0.9275728594469871 0.30071552624547276 4.084995990890504
  position -2.836948434830579 3.0253436212067815 -2.404420359728429
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
  size 3 3
}
Table {
  translation -0.201623 6.66134e-15 -0.863277
  size 0.8 0.3 0.5
  feetSize 0.05 0.05
}
Table {
  translation -0.340768 6.21725e-15 0.671729
  name "table(1)"
  size 0.8 0.3 0.5
  feetSize 0.05 0.05
}
CardboardBox {
  translation 0.516454 0.1 -0.588447
  size 0.2 0.2 0.2
}
CardboardBox {
  translation 1.11091 0.1 0.934764
  name "cardboard box(2)"
  size 0.2 0.2 0.2
}
CardboardBox {
  translation 0.888851 0.1 0.154107
  name "cardboard box(1)"
  size 0.2 0.2 0.2
}
OilBarrel {
  translation -1.15315 0.15 0.801624
  height 0.3
  radius 0.1
}
OilBarrel {
  translation -1.04927 0.15 -1.01841
  name "oil barrel(2)"
  height 0.3
  radius 0.1
}
OilBarrel {
  translation 0.617681 0.15 0.848268
  name "oil barrel(1)"
  height 0.3
  radius 0.1
}
DEF WALL_1 Solid {
  translation 1.5 0.05 0
  scale 3.5 3.5 3.5
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.12549 0.368627 0.729412
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 0.98
      }
    }
  ]
  boundingObject USE WALL_SHAPE
}
DEF WALL_2 Solid {
  translation -1.5 0.05 0
  scale 3.5 3.5 3.5
  children [
    USE WALL_SHAPE
  ]
  name "solid(1)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_3 Solid {
  translation 0 0.05 -1.5
  rotation 0 1 0 1.5708
  scale 3.5 3.5 3.5
  children [
    USE WALL_SHAPE
  ]
  name "solid(2)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_4 Solid {
  translation 0 0.05 1.5
  rotation 0 1 0 1.5708
  scale 3.5 3.5 3.5
  children [
    USE WALL_SHAPE
  ]
  name "solid(3)"
  boundingObject USE WALL_SHAPE
}
Robot {
  translation 0 0.05 0
  rotation 0 -1 0 2.78858
  children [
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 0.207736 0.155104 0.905098
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
    DEF WHEEL1 HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.06 0 0.05
        rotation 0 0 1 1.5708
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0.905882 0.808039 0.20098
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
              subdivision 24
            }
          }
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL2 HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation -0.06 0 0.05
        rotation 0 0 1 1.5708
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL3 HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation 0.06 0 -0.05
        rotation 0 0 1 1.5708
        children [
          USE WHEEL
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL4 HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.06 0 -0.05
        rotation 0 0 1 1.5708
        children [
          USE WHEEL
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF DS_RIGHT DistanceSensor {
      translation -0.03 0 0.1
      rotation 0 1 0 -1.87079
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.184314 0.596078 0.847059
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_right"
    }
    DEF DS_LEFT DistanceSensor {
      translation 0.03 0 0.1
      rotation 0 1 0 -1.27079
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.184314 0.596078 0.847059
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left"
    }
  ]
  boundingObject USE BODY
  physics Physics {
    density -1
    mass 1
  }
  controller "keyboardControl"
}
